dags/vz_moped_component_crashes.py

```diff
@@ -0,0 +1,147 @@
+
+from airflow.models import DAG
+from airflow.operators.docker_operator import DockerOperator
+
+from utils.onepassword import get_env_vars_task
+from utils.slack_operator import task_fail_slack_alert
+
+
+DEPLOYMENT_ENVIRONMENT = os.getenv("ENVIRONMENT")
+secrets_env_prefix = None
+
+if DEPLOYMENT_ENVIRONMENT == "production":
+    secrets_env_prefix = "prod"
+elif DEPLOYMENT_ENVIRONMENT == "staging":
+    secrets_env_prefix = "staging"
+else:
+    secrets_env_prefix = "dev"
+
+
+REQUIRED_SECRETS_SOCRATA = {
+    "SOCRATA_DATASET_CRASHES": {
+        "opitem": "Vision Zero Socrata Export",
+        "opfield": f"{secrets_env_prefix}.SOCRATA_DATASET_CRASHES",
+    },
+    "SOCRATA_DATASET_PEOPLE": {
+        "opitem": "Vision Zero Socrata Export",
+        "opfield": f"{secrets_env_prefix}.SOCRATA_DATASET_PEOPLE",
+    },
+    "SOCRATA_KEY_ID": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeyId",
+    },
+    "SOCRATA_KEY_SECRET": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeySecret",
+    },
+    "SOCRATA_APP_TOKEN": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.appToken",
+    },
+    "HASURA_GRAPHQL_ENDPOINT": {
+        "opitem": "Vision Zero CRIS Import",
+        "opfield": f"{secrets_env_prefix}.HASURA_GRAPHQL_ENDPOINT",
+    },
+    "HASURA_GRAPHQL_ADMIN_SECRET": {
+        "opitem": "Vision Zero CRIS Import",
+        "opfield": f"{secrets_env_prefix}.HASURA_GRAPHQL_ADMIN_SECRET",
+    },
+}
+
+REQUIRED_SECRETS_MOPED_JOIN = {
+    "VZ_HASURA_ENDPOINT": {
+        "opitem": "Vision Zero CRIS Import",
+        "opfield": f"{secrets_env_prefix}.HASURA_GRAPHQL_ENDPOINT",
+    },
+    "VZ_HASURA_ADMIN_SECRET": {
+        "opitem": "Vision Zero CRIS Import",
+        "opfield": f"{secrets_env_prefix}.HASURA_GRAPHQL_ADMIN_SECRET",
+    },
+    "MOPED_HASURA_ENDPOINT": {
+        "opitem": "Moped Hasura Admin",
+        "opfield": "production.Endpoint",
+    },
+    "MOPED_HASURA_ADMIN_SECRET": {
+        "opitem": "Moped Hasura Admin",
+        "opfield": "production.Admin Secret",
+    },
+}
+
+
+docker_image_vz_moped_join = f"atddocker/vz-moped-join:{'production' if DEPLOYMENT_ENVIRONMENT == 'production' else 'development'}"
+
+docker_image_socrata_export = f"atddocker/vz-socrata-export:{'production' if DEPLOYMENT_ENVIRONMENT == 'production' else 'development'}"
+
+
+DEFAULT_ARGS = {
+    "depends_on_past": False,
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 2,
+    "execution_timeout": duration(minutes=60),
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+
+with DAG(
+    catchup=False,
+    dag_id="vz-moped-component-crashes",
+    description="Populate the Moped - crash lookup table and publish the table to Socrata",
+    default_args=DEFAULT_ARGS,
+    schedule_interval="0 0 1 * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
```

Comment:
🕚 midnight on the first of every month

Reply:
Maybe not the first? If we get errors it could get buried in the inevitable 1st of the month AGOL errors we get?